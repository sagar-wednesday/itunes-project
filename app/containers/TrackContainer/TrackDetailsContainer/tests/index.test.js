/**
 *
 * Tests for TrackDetailsContainer container
 *
 *
 */

import React from 'react';
// import { fireEvent } from '@testing-library/dom';
import { renderProvider } from '@utils/testUtils';
import { TrackDetailsContainerTest as TrackDetailsContainer } from '../index';

describe('<TrackDetailsContainer /> container tests', () => {
  let trackDetails;
  let dispatchGetTrackDetailsSpy;

  beforeEach(() => {
    trackDetails = {
      artistName: 'Sia',
      collectionName: 'abc',
      cardImg: 'https://abc.jpg',
      previewUrl: 'https://bcw.mp3'
    };
    dispatchGetTrackDetailsSpy = jest.fn();
  });

  it('should render and match the snapshot', () => {
    const { baseElement } = renderProvider(
      <TrackDetailsContainer dispatchGetTrackDetails={dispatchGetTrackDetailsSpy} trackDetails={trackDetails} />
    );
    expect(baseElement).toMatchSnapshot();
  });
});
