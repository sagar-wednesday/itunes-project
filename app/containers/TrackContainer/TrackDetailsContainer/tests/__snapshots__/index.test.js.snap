// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TrackDetailsContainer /> container tests should render and match the snapshot 1`] = `
<body>
  <div>
    <div
      class="TrackDetailsContainer__Container-gquu6h-0 cXbAcY"
    >
      <div
        class="TrackDetailsContainer__CustomDetailsContainer-gquu6h-1 hcOALb"
      >
        <div
          class="ant-card ant-card-bordered TracksCard__CustomCard-sc-7b0elv-0 bqeuPE"
          data-testid="tracks-card"
          width="15rem"
        >
          <div
            class="ant-card-cover"
          >
            <div
              class="ant-image"
            >
              <img
                alt="image"
                class="ant-image-img"
              />
              <div
                class="ant-image-mask"
              >
                <div
                  class="ant-image-mask-info"
                >
                  <span
                    aria-label="eye"
                    class="anticon anticon-eye"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="eye"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M942.2 486.2C847.4 286.5 704.1 186 512 186c-192.2 0-335.4 100.5-430.2 300.3a60.3 60.3 0 000 51.5C176.6 737.5 319.9 838 512 838c192.2 0 335.4-100.5 430.2-300.3 7.7-16.2 7.7-35 0-51.5zM512 766c-161.3 0-279.4-81.8-362.7-254C232.6 339.8 350.7 258 512 258c161.3 0 279.4 81.8 362.7 254C791.5 684.2 673.4 766 512 766zm-4-430c-97.2 0-176 78.8-176 176s78.8 176 176 176 176-78.8 176-176-78.8-176-176-176zm0 288c-61.9 0-112-50.1-112-112s50.1-112 112-112 112 50.1 112 112-50.1 112-112 112z"
                      />
                    </svg>
                  </span>
                  Preview
                </div>
              </div>
            </div>
          </div>
          <div
            class="ant-card-body"
          >
            <span
              class="TracksCard__CustomTags-sc-7b0elv-7 IhKtr"
            >
              ....
            </span>
            <p
              class="TracksCard__CardHeader-sc-7b0elv-1 gRraVB"
            >
              ....No Title....
            </p>
            <span
              class="TracksCard__CardDescription-sc-7b0elv-2 dMBnzX"
            >
              abc
            </span>
            <div
              class="TracksCard__ButtonContainer-sc-7b0elv-4 kKzPkC"
            >
              <button
                class="ant-btn TracksCard__PlayButton-sc-7b0elv-3 egOMOc"
                type="button"
              >
                <span
                  class="TracksCard__ButtonText-sc-7b0elv-6 crDLcF"
                >
                  Play
                </span>
              </button>
            </div>
            <audio
              data-testid="trackAudio"
              src="https://bcw.mp3"
            />
          </div>
        </div>
        <div
          class="TrackDetailsContainer__CustomTrackExtraDetailsContainer-gquu6h-2 dpnFIq"
        >
          <p
            class="TrackDetailsContainer__CustomTrackExtraDetails-gquu6h-3 gQqRKp"
          >
            <span
              class="TrackDetailsContainer__BoldText-gquu6h-4 cJiIeZ"
            >
              Country: 
            </span>
          </p>
          <p
            class="TrackDetailsContainer__CustomTrackExtraDetails-gquu6h-3 gQqRKp"
          >
            <span
              class="TrackDetailsContainer__BoldText-gquu6h-4 cJiIeZ"
            >
              Artist Name: 
            </span>
            Sia
          </p>
          <p
            class="TrackDetailsContainer__CustomTrackExtraDetails-gquu6h-3 gQqRKp"
          >
            <span
              class="TrackDetailsContainer__BoldText-gquu6h-4 cJiIeZ"
            >
              Genre: 
            </span>
          </p>
        </div>
      </div>
    </div>
  </div>
</body>
`;
